xmake/core/project/target.lua

```diff
@@ -416,9 +416,10 @@ function _instance:get_from_pkgs(name, opt)
         local configinfo = self:pkgconfig(pkg:name())
         -- get values from package components
         -- e.g. `add_packages("sfml", {components = {"graphics", "window"}})`
-        if configinfo and configinfo.components and pkg:components() then
+        local components_default = pkg:components_default()
+        if configinfo and (configinfo.components or components_default) and pkg:components() then
             local components_enabled = hashset.new()
-            for _, comp in ipairs(table.wrap(configinfo.components)) do
+            for _, comp in ipairs(table.join(configinfo.components or {}, components_default)) do
```

Comment:
doesn't that mean that those components can't be disabled? that's not what I meant sorry for the confusion.

`for _, comp in ipairs(table.wrap(configinfo.components or components_default)) do` is more what I suggest

Reply:
? Components that are not set with {compoments=""} and default=true are disabled by default.

you can do not set them.

Reply:
I have improved it, try it again.